work/v1/types.go

```diff
@@ -225,3 +225,68 @@ type ManifestWorkList struct {
 	// Items is a list of manifestworks.
 	Items []ManifestWork `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AppliedManifestWork represents an applied manifestwork on managed cluster. It is placed
+// on managed cluster. An AppliedManifestWork links to a manifestwork on a hub recording resources
+// deployed in the managed cluster.
+// When the agent is removed from managed cluster, cluster-admin on managed cluster
+// can delete appliedmanifestwork to remove resources deployed by the agent.
+type AppliedManifestWork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
```

Comment:
I'd like the name to be forced to include the hub hash so that we can support multiple hubs.  I'd also like to see a force label so we can efficiently select.

Reply:
Perhaps, .metadata.name == `<hubhash>-<manifestwork name>`

Reply:
Will a generate name with manifestwork name as prefix works also? Since we already have hubhash in the spec.
labels makes sense.

Reply:
> Will a generate name with manifestwork name as prefix works also? Since we already have hubhash in the spec.
> labels makes sense.

That makes it possible to accidentally create two appliedmanifestworks for the same underlying manifestwork.  We don't want that to be possible.